outport/process/executionOrder/transactionsExecutionOrder.go

```diff
@@ -0,0 +1,258 @@
+package executionOrder
+
+import (
+	"encoding/hex"
+
+	"github.com/ElrondNetwork/elrond-go-core/core"
+	"github.com/ElrondNetwork/elrond-go-core/core/check"
+	txsSort "github.com/ElrondNetwork/elrond-go-core/core/transaction"
+	"github.com/ElrondNetwork/elrond-go-core/data"
+	"github.com/ElrondNetwork/elrond-go-core/data/block"
+	"github.com/ElrondNetwork/elrond-go-core/data/outport"
+	"github.com/ElrondNetwork/elrond-go-core/data/smartContractResult"
+	"github.com/ElrondNetwork/elrond-go-core/hashing"
+	logger "github.com/ElrondNetwork/elrond-go-logger"
+	"github.com/ElrondNetwork/elrond-go/process"
+)
+
+var log = logger.GetOrCreate("outport/process/executionOrder")
+
+type sorter struct {
+	hasher hashing.Hasher
+}
+
+func NewSorter(hasher hashing.Hasher) (*sorter, error) {
+	if check.IfNil(hasher) {
+		return nil, process.ErrNilHasher
+	}
+
+	return &sorter{
+		hasher: hasher,
+	}, nil
+}
+
+func (s *sorter) PutExecutionOrderInTransactionPool(
+	pool *outport.Pool,
+	header data.HeaderHandler,
+	body data.BodyHandler,
+) error {
+	blockBody, ok := body.(*block.Body)
+	if !ok {
+		log.Warn("s.PutExecutionOrderInTransactionPool cannot cast BodyHandler to *Body")
+		return nil
+	}
+
+	// already sorted
+	transactionsToMe := extractNormalTransactionAndScrsToMe(pool, blockBody, header, true)
+
+	// scheduled to me, already sorted
+	scheduledTransactionsToMe := extractNormalTransactionAndScrsToMe(pool, blockBody, header, false)
+
+	// need to be sorted
+	transactionsFromMe := extractNormalTransactionsAndInvalidFromMe(pool, blockBody, header, true)
+	txsSort.SortTransactionsBySenderAndNonceWithFrontRunningProtectionExtendedTransactions(transactionsFromMe, s.hasher, header.GetPrevRandSeed())
+
+	rewardsTxs := getRewardsTxsFromMe(pool, blockBody, header)
+
+	// scheduled from me, need to be sorted
+	scheduledTransactionsFromMe := extractNormalTransactionsAndInvalidFromMe(pool, blockBody, header, false)
+	txsSort.SortTransactionsBySenderAndNonceWithFrontRunningProtectionExtendedTransactions(scheduledTransactionsFromMe, s.hasher, header.GetPrevRandSeed())
+
+	allTransaction := append(transactionsToMe, scheduledTransactionsToMe...)
+	allTransaction = append(allTransaction, transactionsFromMe...)
+	allTransaction = append(allTransaction, rewardsTxs...)
+	allTransaction = append(allTransaction, scheduledTransactionsFromMe...)
+
+	for idx, tx := range allTransaction {
+		tx.SetExecutionOrder(idx)
+	}
+
+	setOrderSmartContractResults(pool)
+
+	prinPool(pool)
+	return nil
+}
+
+func setOrderSmartContractResults(pool *outport.Pool) {
+	for _, scrHandler := range pool.Scrs {
+		scr, ok := scrHandler.GetTxHandler().(*smartContractResult.SmartContractResult)
+		if !ok {
+			continue
+		}
+
+		tx, found := pool.Txs[string(scr.OriginalTxHash)]
+		if !found {
+			continue
+		}
+
+		scrHandler.SetExecutionOrder(tx.GetExecutionOrder())
```

Comment:
So the scr will have exactly the same execution order as its original tx right? But the order should be: tx, scr and not scr, tx. How you prioritised tx in front of scr in this situation?

Reply:
We started from the idea that always transaction is executed first in this case and after the smart contract results.

Reply:
ok